src/ASM/ASMunstruct.h

```diff
@@ -71,6 +72,10 @@ namespace LR {
   //! \param[in] xi Dimensionless Gauss point coordinates [-1,1]
   void getGaussPointParameters(const LRSpline* spline, RealArray& uGP,
                                int dir, int nGauss, int iel, const double* xi);
+
+  //! \brief Generate thread groups for a LRSpline mesh
+  //! \param[in] lr The LR spline to generate thread groups for
+  ThreadGroups generateThreadGroups(const LR::LRSpline* lr);
```

Comment:
Isn't it better with the signature `void generateThreadGroups(ThreadGroups& threadgroups, const LR::LRSpline* lr);` instead? Then you could forward-declare ThreadGroups instead of including the header, which is included in the sub-class files anyway (superfluous), and you'll avoid copying the ThreadGroups object on return.

Not a big deal, but 'alle monner drar'